storage/src/checker/mod.rs

```diff
@@ -212,7 +212,8 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
                 Ordering::Less => {
                     // continue to the next area
                     leaked.push((current_addr, area_index));
-                    current_addr = next_addr;
+                    current_addr =
+                        LinearAddress::new(next_addr.get()).expect("next_addr is non-zero");
```

Comment:
```
LinearAddress::new(next_addr.get()).expect(...)
```

This should be an anti-pattern. Why is `next_addr` not a `LinearAddress`. The point of the newtype should be to minimize panic points.